src/sql/engine/expr/ob_datum_cast.cpp

```diff
@@ -9437,7 +9437,7 @@ static int float_range_check(const ObCastMode &cast_mode,
   } else {
     IN_TYPE in_val = *(reinterpret_cast<const IN_TYPE*>(in_datum.ptr_));
     IN_TYPE out_val = in_val;
-    if (lib::is_oracle_mode() && 0.0 == in_val) {
+    if (lib::is_oracle_mode() && fabs(in_val - 0.0) < FLT_EPSILON) {
```

Comment:
Why do you use `in_val - 0.0` but not `in_val`?

Reply:
I feel that using `fabs(in_val - 0.0) < FLT_EPSILON` can more clearly express that this part is comparing `in_val` with 0. However, it should be equivalent to directly using `in_val < FLT_EPSILON`. If you think using `in_val < FLT_EPSILON` is better, I can make the modification.

Reply:
Please do it thanks.